example/src/domain/Full.js

```diff
@@ -0,0 +1,115 @@
+import { optional, required, object, minLength, array, message, email, useForm, useFieldValue } from '@kaliber/forms'
+import { requiredWhenInParent, requiredWhenInForm, date, whenInParent } from './machinery/validation'
+import { FormValues, FormTextInput, FormCheckbox, FormObjectField, FormFieldValue, FormArrayField } from './machinery/Form'
+import { Code } from './machinery/Code'
+
+/**
+ * When you study this example, don't forget to check the components in the machinery directory
+ */
+
+const fields = {
+  naam: optional,
+  email: [required, email],
+  geboortedatum: [required, date],
+  kortingscode: optional,
+  betaalNu: required,
+  betaalInfo: object({
+    andereNaam: optional,
+    rekeninghouder: requiredWhenInParent(x => x.andereNaam),
+    rekeningnummer: [requiredWhenInForm(x => x.betaalNu), minLength(9)],
+  }),
+  extraKaartjes: array(
+    (x, { form }) => form.kortingscode && minLength(1)(x) && message('kortingMoetMetVrienden'),
```

Comment:
Dit lijkt me een custom validatie, maar ik snap niet precies wat `message` hier doet. Ik vind de string ook nergens anders terug. 

Edit: Ik kan het wel terug beredeneren vanuit de andere validatie regels, maar daar zijn de booleans true wanneer de waarde invalid is, terwijl form.kortingscode dit pattern niet aan lijkt te houden. Ook de beschrijving in de readme wordt me nu duidelijk, maar daar zou een stukje toelichting toch niet verkeerd zijn denk ik.

Reply:
`message` geeft volgensmij de validatie/error string terug, zou dit niet in de naam terug moeten komen? Nu kwam het op mij over als een extra validatieregel

Reply:
Goed punt. Ik zal in de readme een extra stuk voor validatie schrijven.

Wat betreft `message`, ik snap wat je bedoelt. Misschien moet ik er `validationMessage` of `validationError` van maken. Een andere mogelijkheid is om de import een beetje uit elkaar te trekken. Nu staat alles bij elkaar in `@kaliber/forms`, maar is misschien geen gek idee om de validatie uit `@kaliber/forms/validation` te halen. Dan zou het wel gewoon `error` of `message` kunnen heten.

Hmm, misschien is het geen gek om de functie `invalid` te noemen aangezien de property in de snapshot ook zo heet.

Reply:
Heb em voor nu `error` genoemd